### DISCUSSION TITLE
Restrictions on RAM memory usage?

### DISCUSSION THREAD

**Discussion by IsotoCedex** [Universite-Gustave-Eiffel/NoiseModelling#621] [Q&A]
Good morning @NoiseModellingTeam,

Is it possible that Noise Modeling restricts the use of RAM memory?

In my case, I use a computer with the following characteristics:
- Processor: Intel(R) Core(TM) i5-8500 CPU @ 3.00GHz 3.00 GHz
- RAM memory: 64.0 GB (63.8 GB usable)
- Windows 64-bit operating system, x64-based processor

However, during the calculation process, the consumed RAM resources are always kept at 16 GB.

![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/87642224/d6a2a1ba-eb68-4778-8fde-8664c6d3f92c)

Is it possible to increase RAM utilization? Would it speed up the calculation?

Thank you!

**Comment by nicolas-f** [comment]
Hello,

Yes NoiseModelling is running under Java virtual machine so the memory usage must be predefined before the launch.

Currently under windows there is no maximum value set so it must be a system wide setting ?

You can overwrite this setting by providing a new parameter in the file `bin/startup_windows.bat` 

here for 38 gigabytes

```powershell
"%RUN_JAVA%" %JAVA_OPTS% -Xmx38000m -Xms1024m -DGEOSERVER_DATA_DIR="%GEOSERVER_DATA_DIR%" -Djava.awt.headless=true -DSTOP.PORT=8079 -DSTOP.KEY=geoserver -jar start.jar
```

**Reply by IsotoCedex** [reply]
Thank you so much @nicolas-f ,

I have verified that the system has no restrictions, and I have modified the file  `startup_windows.bat` for 50GB.

`"%RUN_JAVA%" %JAVA_OPTS% -Xmx51200m -Xms1024m -DGEOSERVER_DATA_DIR="%GEOSERVER_DATA_DIR%" -Djava.awt.headless=true -DSTOP.PORT=8079 -DSTOP.KEY=geoserver -jar start.jar`

Now NoiseModelling is using more than double RAM than before (42 GB compared to the 16GB it used before).

![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/87642224/9130e42a-6374-42e1-949b-f9a8dcedca6a)

I assume that it does not use the 50 GB that I have put in the file, because it does not have enough CPU capacity. It is a limited capacity processor (i5).

**Reply by IsotoCedex** [reply]
Finally the maximum configured limit (50 GB) is reached. In the model I am calculating, it has been reached from 6% of the calculation process.

![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/87642224/5457ce8d-e100-4390-ba29-6fbd86562b19)

**Reply by nicolas-f** [reply]
You must have set a large propagation distance if it was an issue with 17 gb of ram ?

**Reply by IsotoCedex** [reply]
Yes, I use a very large propagation distance (10 km), a large number of sources and a very large domain. You can see it on the link:
https://documentacion.sea-acustica.es/publicaciones/Cuenca23/Abs_54.pdf

I can do the calculation when using a light mesh of receivers (200x200 m) and light DEM data (300x300 m). It is done with 16 GB, although it lasts a few days (2-3).

When I try to densify the meshes, NoiseModelling crashes in the middle of the calculation, due to a memory failure.

The thing is that I realized that no more than 16GB was used, when I have 64 GB available on that computer.

Now I am calculating a new model, with a 100x100m receivers mesh, a 50x50m DEM mesh and a 10 km propagation distance.

I'll tell you if it succeeds or fails :)

Thank you!!
